--- a/drivers/soc/rockchip/fiq_debugger/fiq_debugger.c
+++ b/drivers/soc/rockchip/fiq_debugger/fiq_debugger.c
@@ -1059,3 +1059,3 @@
 static void fiq_debugger_fiq(struct fiq_glue_handler *h,
 		void *regs, void *svc_sp)
 {
-	struct fiq_debugger_state *state =
-		container_of(h, struct fiq_debugger_state, handler);
-	unsigned int this_cpu = THREAD_INFO(svc_sp)->cpu;
+	struct fiq_debugger_state *state =
+		container_of(h, struct fiq_debugger_state, handler);
+	unsigned int this_cpu = smp_processor_id();
 	bool need_irq;
 
 	need_irq = fiq_debugger_handle_uart_interrupt(state, this_cpu, regs,
 		svc_sp);
 	if (need_irq)
 		fiq_debugger_force_irq(state);
 }
--